ably/realtime_client_test.go

```diff
@@ -47,6 +47,83 @@ func TestRealtime_RealtimeHost(t *testing.T) {
 	}
 }
 
+func TestRealtime_RTN17_HostFallback(t *testing.T) {
+
+	setUpWithEOF := func() (app *ablytest.Sandbox, client *ably.Realtime, doEOF chan struct{}, rec *MessageRecorder) {
+		doEOF = make(chan struct{}, 1)
+		rec = NewMessageRecorder()
+		app, client = ablytest.NewRealtime(
+			ably.WithAutoConnect(false),
+			ably.WithDial(func(protocol string, u *url.URL, timeout time.Duration) (ably.Conn, error) {
+				c, err := rec.Dial(protocol, u, timeout)
+				return protoConnWithFakeEOF{Conn: c, doEOF: doEOF}, err
+			}))
+		return
+	}
+
+	// todo - we can add check after getting disconnected and connected again
+	t.Run("RTN17a: First attempt should be on default host first", func(t *testing.T) {
+		t.Parallel()
+		app, client, _, recorder := setUpWithEOF()
+		defer safeclose(t, ablytest.FullRealtimeCloser(client), app)
+		err := ablytest.Wait(ablytest.ConnWaiter(client, client.Connect, ably.ConnectionEventConnected), nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+		connectionState := client.Connection.State()
+		if connectionState != ably.ConnectionStateConnected {
+			t.Fatalf("expected %v; got %v", ably.ConnectionStateConnected, connectionState)
+		}
+		retriedHost := recorder.URLs()[0].Hostname()
+		expectedHost := "sandbox-realtime.ably.io"
+		if retriedHost != expectedHost {
+			t.Fatalf("expected %v; got %v", expectedHost, retriedHost)
+		}
+	})
+
+	t.Run("RTN17b: Fallback behaviour", func(t *testing.T) {
+		t.Parallel()
+		t.Run("does not apply when the default custom endpoint is used", func(t *testing.T) {
+			t.Parallel()
+			rec := NewMessageRecorder()
+			app, client := ablytest.NewRealtime(
+				ably.WithAutoConnect(false),
+				ably.WithRealtimeHost("un.reachable.host.example.com"),
+				ably.WithDial(rec.Dial))
+			defer safeclose(t, ablytest.FullRealtimeCloser(client), app)
+			err := ablytest.Wait(ablytest.ConnWaiter(client, client.Connect, ably.ConnectionEventDisconnected), nil)
+			if err == nil {
+				t.Fatal("Should return error for unreachable host")
+			}
+			// Explicit check to not retry on fallbacks
+		})
+
+		t.Run("apply when HTTP client is using same fallback endpoint and default realtime endpoint not overriden", func(t *testing.T) {
+
+		})
+
+		t.Run("does not apply when environment is overriden and fallback not specified", func(t *testing.T) {
+
+		})
+
+		t.Run("apply when environment is overriden and fallback specified, the fallback is used", func(t *testing.T) {
+
+		})
+	})
+
+	t.Run("RTN17c: Verifies internet connection is active in case of error necessitating use of an alternative host", func(t *testing.T) {
```

Comment:
Could you mark empty tests with `t.Skip`? This way, if we forget to finish here, it will show up in test runs.

(I realize this is realtime and thus probably part of #359.)

Reply:
Yes, it's a part of realtime fallback and I have updated it